pandas/src/algos_take_helper.pxi

```diff
@@ -4947,3 +4947,22 @@ def take_2d_multi_object_object(ndarray[object, ndim=2] values,
                     out[i, j] = fv
                 else:
                     out[i, j] = values[idx, idx1[j]]
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def take_bounds_check(int64_t[:] indexer, int64_t[:] out, int64_t n):
+    cdef:
```

Comment:
I know
I would eliminate many_convert_indixes and just use this (make it a cpdef) if u can